Mathlib/Algebra/Module/LinearMap/Basic.lean

```diff
@@ -649,13 +669,11 @@ namespace DistribMulActionHom
 variable [Semiring R] [AddCommMonoid M] [AddCommMonoid M₂] [Module R M] [Module R M₂]
 
 /-- A `DistribMulActionHom` between two modules is a linear map. -/
-@[coe]
-def toLinearMap (fₗ : M →+[R] M₂) : M →ₗ[R] M₂ :=
-  { fₗ with }
-#align distrib_mul_action_hom.to_linear_map DistribMulActionHom.toLinearMap
+instance : LinearMapClass (M →+[R] M₂) R M M₂ where
+  map_smulₛₗ := map_smul
 
-instance : CoeTC (M →+[R] M₂) (M →ₗ[R] M₂) :=
-  ⟨toLinearMap⟩
+instance : CoeTC (M →+[R] M₂) (M →ₗ[R] M₂) where
+  coe f := SemilinearMapClass.semilinearMap f
```

Comment:
Is there a reason why this is `CoeTC` and not just `Coe` here? (The documentation in core says that `CoeTC` shouldn't be used directly...)

Reply:
This was already there! docs#[DistribMulActionHom.instCoeTCDistribMulActionHomToMonoidToMonoidWithZeroToAddMonoidToDistribMulActionToAddMonoidToDistribMulActionLinearMapIdToNonAssocSemiring](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/LinearMap/Basic.html#DistribMulActionHom.instCoeTCDistribMulActionHomToMonoidToMonoidWithZeroToAddMonoidToDistribMulActionToAddMonoidToDistribMulActionLinearMapIdToNonAssocSemiring)

Reply:
I realize this was there already (and that there is quite a bit of this going on in mathlib), but I'm still interested in the answer!

(Also, how about naming the instance? :-) )

Reply:
According to the blame view, this was done in #6786 to make sure this only applies in the specific case of going from `DistribMulActionHom` to `LinearMap`. Maybe it should be `CoeHead` instead?

Reply:
I see. Since it was done very deliberately I'm happy to leave it be!

Reply:
I have named the instances. I think we should leave the `CoeHead` changes to another PR where we do this more systematically for the whole of mathlib.